core/domain/user_jobs_one_off_test.py

```diff
@@ -31,6 +31,109 @@
+    USER_B_USERNAME = 'b'
+    USER_C_EMAIL = 'c@example.com'
+    USER_C_USERNAME = 'c'
+    USER_D_EMAIL = 'd@example.com'
+    USER_D_USERNAME = 'd'
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = user_jobs_one_off.UserContributionsOneOffJob.create_new()
+        user_jobs_one_off.UserContributionsOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+            1)
+        self.process_and_flush_pending_tasks()
+
+    def setUp(self):
+        super(UserContributionsOneOffJobTests, self).setUp()
+
+        self.signup(self.USER_A_EMAIL, self.USER_A_USERNAME)
+        #User A has no created or edited explorations 
+        #User B has one created exploration 
+        #User C has one edited exploration 
+        # User D has created an exploration and then edited it. 
+        #(This is used to check that there are no duplicate 
+        #entries in the contribution lists.)
+        self.user_a_id = self.get_user_id_from_email(self.USER_A_EMAIL)
+        self.signup(self.USER_B_EMAIL, self.USER_B_USERNAME)
+        self.user_b_id = self.get_user_id_from_email(self.USER_B_EMAIL)
+        self.signup(self.USER_C_EMAIL, self.USER_C_USERNAME)
+        self.user_c_id = self.get_user_id_from_email(self.USER_C_EMAIL)
+        self.signup(self.USER_D_EMAIL, self.USER_D_USERNAME)
+        self.user_d_id = self.get_user_id_from_email(self.USER_D_EMAIL)
+
+        self.save_new_valid_exploration(
+            self.EXP_ID_1, self.user_b_id, end_state_name='End')
+
+        exp_services.update_exploration(self.user_c_id, self.EXP_ID_1, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+            }], 'Test edit')
+
+        self.save_new_valid_exploration(
+            self.EXP_ID_2, self.user_d_id, end_state_name='End')
+
+        exp_services.update_exploration(self.user_d_id, self.EXP_ID_2, [{
+            'cmd': 'edit_exploration_property',
+            'property_name': 'objective',
+            'new_value': 'the objective'
+            }], 'Test edit')
+
+    def test_null_case(self):
+        """Tests the case where user has no created or edited explorations."""
+
+        self._run_one_off_job()
+        user_a_contributions_model = user_models.UserContributionsModel.get(
+            self.user_a_id, strict=False)
+        self.assertEqual(user_a_contributions_model.created_explorations, [])
+        self.assertEqual(user_a_contributions_model.edited_explorations, [])
+
+
+    def test_created_exp(self):
+        """Tests the case where user has created (and therefore edited) 
+        an exploration."""
+
+        self._run_one_off_job()
+
+        user_b_contributions_model = user_models.UserContributionsModel.get(
+            self.user_b_id)
+
+        self.assertEqual(
```

Comment:
Should check their lengths as well. In fact, why not just check equality of the entire list?